circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -189,7 +222,13 @@ public void updateMetadata(
       if (!partitionsToAlter.isEmpty()) {
         LOG.info("Altering {} existing partitions.", partitionsToAlter.size());
         try {
-          client.alter_partitions(replicaDatabaseName, replicaTableName, partitionsToAlter);
+          int counter = 0;
+          for (List<Partition> sublist : Lists.partition(partitionsToAlter, partitionBatchSize)) {
+            int start = counter * partitionBatchSize;
+            LOG.info("Altering partitions {} through {}", start, start + sublist.size() - 1);
+            client.alter_partitions(replicaDatabaseName, replicaTableName, partitionsToAlter);
```

Comment:
shouldn't this be `sublist` instead of `partitionsToAlter`.

`client.alter_partitions(replicaDatabaseName, replicaTableName, partitionsToAlter);`

Reply:
good catch, thanks!